annsa/data.py

```diff
@@ -0,0 +1,143 @@
+import pandas as pd
+from annsa.template_sampling import make_random_spectrum
+import os
+
+class spectrum_data(object):
+	"""docstring for spectrum_data"""
+	def __init__(self, filename, folder=False, ae=False, csv=False, online=False):
+		"""
+		Generates a data object. 
+
+		Parameters: 
+		-----------
+		filename : string
+			This is the path including the name of the file
+			that contains your raw data.
+		folder : boolean
+			This says whether filename leads to a folder of
+			data, or a single dataset. 
+			Default is false
+		ae : boolean
+			This says whether the data should be processed 
+			for autoencoders or not. 
+			Default is false.
+		csv : boolean
+			If you are inputting data from a csv file this 
+			should be true. 
+			Default is false.
+		online : boolean, optional
+			Whether or not the data set should be sampled by
+			using online_data_augmentation or not.
+		"""
+		super(spectrum_data, self).__init__()
+
+		self.keys = ['Isotope', 'Distance', 'Height', 'Shielding', 'Areal Density', 'FWHM', 'Spectrum']
+		self.filename = filename
+		self.dataframe = pd.DataFrame()
+
+	def add_column(self):
+		"""
+		Adds a column of data to the spectrum dataframe. 
+		"""
+		pass
+
+	def add_row(self):
+		"""
+		Adds a row of data to the spectrum dataframe.
+		This is useful if you want to add more data to 
+		an existing dataframe.
+		"""
+		#gets the keys from the file name
+		values = parse_fname(self.filename)
+
+		#read in the data from the file
+		#...
+		#...
+
+		value.append(spectrum) #spectrum is a list
+
+		dictionary = {}
+
+		for key in self.keys:
+			i = 0
+			dictionary[key] = values[i]
+			i += 1
+
+		row = pd.DataFrame(dictionary)
+		
+		#adds the row to the bottom of the dataframe
+		frames = [self.dataframe, row]
+		self.dataframe = pd.concat(frames)
+		
+		pass
+
+	def parse_fname(self, filename):
+		"""
+		Separates the name of a single file into key
+		identifiers based on the standard annsa naming
+		convention.
+
+		Parameters: 
+		-----------
+		filename : string
+			Name of file containing raw spe data.
+
+		Returns: 
+		--------
+		parameters : list of strings
+			This list of strings will become the 
+			data for the preset column headers 
```

Comment:
This docstring should be completely rewritten because it does not reflect the actual use of the function. The standard dataframe will have columns with keys stored in `self.keys`. The filename contains data that will be inserted in a new row of the dataframe.